airflow/providers/google/cloud/hooks/cloud_memorystore.py

```diff
@@ -499,3 +502,434 @@ def update_instance(
+        self,
+        location: str,
+        instance_id: str,
+        instance: Union[Dict, cloud_memcache.Instance],
+        project_id: str,
+        retry: Optional[Retry] = None,
+        timeout: Optional[float] = None,
+        metadata: Optional[Sequence[Tuple[str, str]]] = None,
+    ):
+        """
+        Creates a Memcached instance based on the specified tier and memory size.
+
+        By default, the instance is accessible from the project's `default network
+        <https://cloud.google.com/compute/docs/networks-and-firewalls#networks>`__.
+
+        :param location: The location of the Cloud Memorystore instance (for example europe-west1)
+        :type location: str
+        :param instance_id: Required. The logical name of the Memcached instance in the customer project
+            with the following restrictions:
+
+            -  Must contain only lowercase letters, numbers, and hyphens.
+            -  Must start with a letter.
+            -  Must be between 1-40 characters.
+            -  Must end with a number or a letter.
+            -  Must be unique within the customer project / location
+        :type instance_id: str
+        :param instance: Required. A Memcached [Instance] resource
+
+            If a dict is provided, it must be of the same form as the protobuf message
+            :class:`~google.cloud.memcache_v1beta2.types.cloud_memcache.Instance`
+        :type instance: Union[Dict, google.cloud.memcache_v1beta2.types.cloud_memcache.Instance]
+        :param project_id: Project ID of the project that contains the instance. If set
+            to None or missing, the default project_id from the GCP connection is used.
+        :type project_id: str
+        :param retry: A retry object used to retry requests. If ``None`` is specified, requests will not be
+            retried.
+        :type retry: google.api_core.retry.Retry
+        :param timeout: The amount of time, in seconds, to wait for the request to complete. Note that if
+            ``retry`` is specified, the timeout applies to each individual attempt.
+        :type timeout: float
+        :param metadata: Additional metadata that is provided to the method.
+        :type metadata: Sequence[Tuple[str, str]]
+        """
+        client = self.get_conn()
+        parent = path_template.expand(
+            "projects/{project}/locations/{location}", project=project_id, location=location
+        )
+        instance_name = CloudMemcacheClient.instance_path(project_id, location, instance_id)
+        try:
+            instance = client.get_instance(
+                name=instance_name, retry=retry, timeout=timeout, metadata=metadata
+            )
+            self.log.info("Instance exists. Skipping creation.")
+            return instance
+        except NotFound:
+            self.log.info("Instance not exists.")
+
+        if isinstance(instance, dict):
+            instance = ParseDict(instance, cloud_memcache.Instance())
```

Comment:
When I was trying to run it i got the following error:
(line number is different because I rebased on the latest master locally)


```
[2020-09-25 12:32:12,847] {cloud_memorystore.py:680} INFO - Instance not exists.
[2020-09-25 12:32:12,849] {taskinstance.py:1337} ERROR - 'DESCRIPTOR'
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/proto/message.py", line 520, in __getattr__
    pb_type = self._meta.fields[key].pb_type
KeyError: 'DESCRIPTOR'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/airflow/airflow/models/taskinstance.py", line 1076, in _run_raw_task
    self._prepare_and_execute_task_with_callbacks(context, task)
  File "/opt/airflow/airflow/models/taskinstance.py", line 1198, in _prepare_and_execute_task_with_callbacks
    result = self._execute_task(context, task_copy)
  File "/opt/airflow/airflow/models/taskinstance.py", line 1243, in _execute_task
    result = task_copy.execute(context=context)
  File "/opt/airflow/airflow/providers/google/cloud/operators/cloud_memorystore.py", line 1291, in execute
    metadata=self.metadata,
  File "/opt/airflow/airflow/providers/google/common/hooks/base_google.py", line 373, in inner_wrapper
    return func(self, *args, **kwargs)
  File "/opt/airflow/airflow/providers/google/cloud/hooks/cloud_memorystore.py", line 683, in create_instance
    instance = ParseDict(instance, cloud_memcache.Instance())
  File "/usr/local/lib/python3.7/site-packages/google/protobuf/json_format.py", line 452, in ParseDict
    parser.ConvertMessage(js_dict, message)
  File "/usr/local/lib/python3.7/site-packages/google/protobuf/json_format.py", line 476, in ConvertMessage
    message_descriptor = message.DESCRIPTOR
  File "/usr/local/lib/python3.7/site-packages/proto/message.py", line 525, in __getattr__
    raise AttributeError(str(ex))
AttributeError: 'DESCRIPTOR'
```

Reply:
I created issue for it: https://github.com/googleapis/python-memcache/issues/19
`ParseDict` and `MessageToDict` have problems with parsing `google.cloud.memcache_v1beta2.types.cloud_memcache.Instance`